compiler/rustc_borrowck/src/diagnostics/conflict_errors.rs

```diff
@@ -3112,12 +3112,22 @@ impl<'infcx, 'tcx> MirBorrowckCtxt<'_, 'infcx, 'tcx> {
             drop_span, borrow_span
         );
 
+        // `TerminatorKind::Return`'s span (the `drop_span` here) `lo` can be subtly wrong and point
+        // at a single character after the end of the function. This is somehow relied upon in
+        // existing diagnostics, and changing this in `rustc_mir_build` makes diagnostics worse in
+        // general. We fix these here.
+        let lo = drop_span.lo().0;
+        let end_of_function = if lo == drop_span.hi().0 && lo > 0 {
```

Comment:
can you just check that `drop_span.is_empty() && drop_span != DUMMY_SP`

Reply:
I can, and did, and then traced more to convince myself `DUMMY_SP` can't appear here. So I just pushed the `is_empty()` change while listening to MIKE.

Reply:
based